lib/root/widgets/cart_icon.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:badges/badges.dart' as badges;
+import 'package:flex_storefront/flex_ui/tokens/sizes.dart';
+import 'package:flex_storefront/root/cubit/cart_icon_cubit.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:icons_plus/icons_plus.dart';
+
+class CartIcon extends StatelessWidget {
+  const CartIcon({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final count =
+        context.select((CartIconCubit cubit) => cubit.state.totalItems);
```

Comment:
We always face the dilemma of "autonomous widgets" receiving their properties in params versus "connected widgets" that get their data from state management...

If we are in the mindset of offering a design system with a set of eCom pre-defined widgets, maybe the "autonomous approach" is better (unless the CartIconCubit is also a part of this design system, i.e. the cubit and the widget go as a pair).

Reply:
Totally agree. Was thinking this would be a future discussion once we decide how autonomous the Flex UI is vs. being "dumb" and passing in all params.